### PR TITLE
simplify-mark-framework-results-script

### PR DESCRIPTION
**Pull request by benvand**
Simplify script
Avoid extraneous/ confusing initialisation of service_counter
Add more comments

Makes the script a bit more verbose but I think more readable. I've annotated each of the pass/ fail conditions with a comment saying what we're checking for and what that results in.

### CODE REVIEW

File: dmscripts/mark_definite_framework_results.py [hunk 1]
```diff
@@ -108,42 +121,62 @@ def mark_definite_framework_results(
...
+            # Congratulations supplier, you pass!
+            pass_supplier(supplier_id, framework_slug, updated_by, supplier_framework, client, logger, dry_run=dry_run)
+            continue
+
+        # The supplier didn't pass, but check for, and default to, a discretionary pass
+        if declaration_discretionary_pass_schema:
+            supplier_passes_discretionary_validation = _passes_validation(
                 supplier_framework["declaration"],
-                declaration_definite_pass_schema,
+                declaration_discretionary_pass_schema,
```
Comment (lfdebrux) [inline dmscripts/mark_definite_framework_results.py]: Is this a change in the logic of the script? are we no longer doing definite passes?

File: tests/test_mark_definite_framework_results.py [hunk 2]
```diff
@@ -38,21 +38,27 @@ def _get_draft_services(self):
     @pytest.mark.parametrize(
         # we can very easily parametrize this into the 16 possible combinations of these flags - the results for the
         # first three flags should be identical and it's very easy to flip some of the assertions for the dry_run mode
-        "reassess_passed,reassess_failed,reassess_failed_ds,dry_run",
+        "reassess_passed_suppliers,reassess_failed_suppliers,reassess_failed_suppliers_ds,dry_run",
```
Comment (lfdebrux) [inline tests/test_mark_definite_framework_results.py]: Could we rename `reassess_failed_suppliers_ds` to `reassess_failed_draft_services`?

File: tests/test_mark_definite_framework_results.py [hunk 3]
```diff
@@ -38,21 +38,27 @@ def _get_draft_services(self):
...
         tuple(product(*repeat((False, True,), 4))),
     )
-    def test_with_ds_schema(self, reassess_passed, reassess_failed, reassess_failed_ds, dry_run,):
+    def test_with_ds_schema(
```
Comment (lfdebrux) [inline tests/test_mark_definite_framework_results.py]:
```suggestion
    def test_with_draft_services_schema(
```

File: tests/test_mark_definite_framework_results.py [hunk 4]
```diff
@@ -104,27 +116,27 @@ def test_without_ds_schema(self, reassess_passed, reassess_failed, reassess_fail
 
     @pytest.mark.parametrize(
         # see above explanation of parameterization
-        "reassess_passed,reassess_failed,reassess_failed_ds,dry_run",
+        "reassess_passed_suppliers,reassess_failed_suppliers,reassess_failed_suppliers_ds,dry_run",
         tuple(product(*repeat((False, True,), 4))),
     )
     def test_no_baseline_schema(
```
Comment (lfdebrux) [inline tests/test_mark_definite_framework_results.py]:
```suggestion
    def test_no_discretionary_pass_schema(
```

**Comment by lfdebrux** [comment]
I left a few comments, I like the thinking behind the changes, I don't fully understand the script though so it might be good to get a second pair of eyes to check there have been no changes to the script logic.

File: dmscripts/mark_definite_framework_results.py [hunk 5]
```diff
@@ -108,42 +121,62 @@ def mark_definite_framework_results(
...
-                    client.set_framework_result(supplier_id, framework_slug, False, updated_by)
-                else:
-                    logger.debug("\tUnchanged result - not re-setting")
+                    dry_run=dry_run
+                )
+                continue
+        # a result of DISCRETIONARY should pretty much never overwrite an already-made decision
+        log_message = "\tResult: DISCRETIONARY%s"
+        if supplier_framework["onFramework"] is not None:
+            log_message += " (but leaving as {})".format("PASS" if supplier_framework["onFramework"] else "FAIL")
```
Comment (lfdebrux) [inline dmscripts/mark_definite_framework_results.py]: I can't see where this log_message is sent to the logger

File: dmscripts/mark_definite_framework_results.py [hunk 6]
```diff
@@ -75,11 +75,11 @@ def mark_definite_framework_results(
...
     framework_slug,
     declaration_definite_pass_schema,
-    declaration_baseline_schema=None,
+    declaration_discretionary_pass_schema=None,
     service_schema=None,
     dry_run=True,
-    reassess_passed=False,
-    reassess_failed=False,
+    reassess_passed_suppliers=False,
+    reassess_failed_suppliers=False,
```
Comment (katstevens) [inline dmscripts/mark_definite_framework_results.py]: These don't match the top level script?

**Comment by benvand** [comment]
@lfdebrux @katstevens Thanks for the comments, good catches, all updated now, tests done as a separate commit.

**Comment by benvand** [comment]
(👋 Bye Mandrill!)

**Pull request merged**